src/ray/gcs/gcs_server/gcs_job_manager.cc

```diff
@@ -111,6 +116,10 @@ void GcsJobManager::HandleAddJob(rpc::AddJobRequest request,
       gcs_table_storage_->JobTable().Put(job_id, mutable_job_table_data, on_done);
   if (!status.ok()) {
     on_done(status);
+  } else {
+    // Intentionally not checking return value, since the function could be invoked for
+    // multiple times and requires idempotency (i.e. due to retry).
+    running_job_ids_.insert(job_id);
```

Comment:
I think we can put this insert into the on_done lambda. The logic for the status OK check is: "regardless of gcs storage succeeded or not we will carry on to serve the job" so it's appropriate to also add to running_job_ids into the callback.

Reply:
Sure, updated.